### ISSUE TITLE
menu item not appearing for VSTS project in VS2019

### ISSUE THREAD

**Issue by amandal1810** [reduckted/GitWebLinksVS2019#8]
hi, i really want this extension to work!

its working for one VSTS project in Visual Studio 2019 (say repo-A). But its not working for another VSTS project in Visual Studio 2019 (say repo-B). The menu command appears for the first time after installation for repo-B. But clicking it once in repo-B solution removes the command from the menu. And the git web link url does not even get copied.
Any help?

**Comment by reduckted** [comment]
Hi, that's really strange. I can't reproduce this behavior on the VSTS repository that I use for testing, but as you say, it works OK for you on one repository and not the other.

I've just published version 1.4.0 which has an option to turn on debug logging (under Tools -> Options -> Git Web Links)

![image](https://user-images.githubusercontent.com/10321525/80939553-5f77b600-8e20-11ea-8b64-e2901d441979.png)

Turn that option on, then open a solution and try to copy the link. In the _Output Window_, select the _Git Web Links_ output. 

![image](https://user-images.githubusercontent.com/10321525/80939739-e9278380-8e20-11ea-82cf-9a94b938778e.png)

You should see something like this:

```
Solution opened. Searching for Git information...
Searching for Git repository from solution directory 'D:\Code\GitHub\GitWebLinks'.
Found Git repository at 'D:\Code\GitHub\GitWebLinks'.
Using remote 'git@github.com:reduckted/GitWebLinks.git'.
Remote URL 'git@github.com:reduckted/GitWebLinks.git' is not match for Azure DevOps.
Remote URL 'git@github.com:reduckted/GitWebLinks.git' is not match for Bitbucket.
Remote URL 'git@github.com:reduckted/GitWebLinks.git' is not match for Visual Studio Team Services.
Remote URL 'git@github.com:reduckted/GitWebLinks.git' is not match for Bitbucket.
Remote URL 'git@github.com:reduckted/GitWebLinks.git' is a match for GitHub.
Git information found for remote 'git@github.com:reduckted/GitWebLinks.git' at 'GitHub'.
Command 'Copy Link to Solution Explorer Item' is enabled with handler 'GitHub'.
Command 'Copy Link to Solution Explorer Item' is enabled with handler 'GitHub'.
Command 'Copy Link to Solution Explorer Item' is enabled with handler 'GitHub'.
URL for GitHub is 'https://github.com/reduckted/GitWebLinks/blob/609a4ea93bbe6b51eb93804a5c304a52fe10a8dc/source/GitWebLinks/Git/GitInfo.vb'
Setting clipboard text to 'https://github.com/reduckted/GitWebLinks/blob/609a4ea93bbe6b51eb93804a5c304a52fe10a8dc/source/GitWebLinks/Git/GitInfo.vb'.
```

If you can email the output to me, that will help me to work out what might be causing the problem. 

Thanks!

**Comment by amandal1810** [comment]
thanks a lot @reduckted for replying!

redacting and copy-pasting the logs here for reference of others. I have mailed you the unredacted version of the logs.

```
Available link handlers:
 * Azure DevOps
 * Bitbucket
 * Visual Studio Team Services
 * Bitbucket
 * GitHub
Solution opened. Searching for Git information...
Searching for Git repository from solution directory 'C:\work\code\{redacted}'.
Found Git repository at 'C:\work\code\{redacted}'.
Using remote 'https://{redacted}.visualstudio.com/DefaultCollection/{redacted}/_git/{redacted}'.
Remote URL 'https://{redacted}.visualstudio.com/DefaultCollection/{redacted}/_git/{redacted}' is not match for Azure DevOps.
Remote URL 'https://{redacted}.visualstudio.com/DefaultCollection/{redacted}/_git/{redacted}' is not match for Bitbucket.
Remote URL 'https://{redacted}.visualstudio.com/DefaultCollection/{redacted}/_git/{redacted}' is not match for Visual Studio Team Services.
Remote URL 'https://{redacted}.visualstudio.com/DefaultCollection/{redacted}/_git/{redacted}' is not match for Bitbucket.
Remote URL 'https://{redacted}.visualstudio.com/DefaultCollection/{redacted}/_git/{redacted}' is not match for GitHub.
No Git link information found.

```

i believe the issue lies in [this](https://github.com/reduckted/GitWebLinks/blob/master/source/GitWebLinks/Links/Handlers/VisualStudioTeamServicesHandler.vb) file and in this line:
`Private Shared ReadOnly HttpsPattern As New Regex("^https:\/\/(?<username>[^.]+)\.visualstudio\.com(?:\/(?<collection>[^\/]+))?\/_git\/.+$")`

looks like the regex does not consider `DefaultCollection` in the git url, because of which the extension is not able to determine the git link handler.

screenshots from regexr.com:

![image](https://user-images.githubusercontent.com/8520454/81155674-d2a93580-8fa2-11ea-9061-5be75bb73627.png)

![image](https://user-images.githubusercontent.com/8520454/81155768-ebb1e680-8fa2-11ea-84af-67beebe13636.png)

**Comment by reduckted** [comment]
@amandal1810 Thanks for you help with this.

You're correct that the regular expression was the problem. I've fixed this in d75f729eebf5f2c1053af1bc098bda89092c50ff and I've published v1.4.1 to the marketplace. It should be available now. 🎉
